src/ServiceControl.Config/Commands/ForceUpgradeServiceControlInstanceCommand.cs

```diff
@@ -0,0 +1,128 @@
+
+        //HINT: Force upgrade is available only primary v4 instance, running on RavenDB 3.5
+        return instance != null && instance.Version.Major == 4 && instance.PersistenceManifest.Name == StorageEngineNames.RavenDB35;
+    }
+    public override async Task ExecuteAsync(ServiceControlAdvancedViewModel model)
+    {
+        if (LicenseChecks)
+        {
+            var licenseCheckResult = serviceControlInstaller.CheckLicenseIsValid();
+            if (!licenseCheckResult.Valid)
+            {
+                await windowManager.ShowMessage("LICENSE ERROR", $"Upgrade could not continue due to an issue with the current license. {licenseCheckResult.Message}.  Contact contact@particular.net", hideCancel: true);
+                return;
+            }
+        }
+
+        if (!ForcedUpgradeAllowed(model))
+        {
+            await windowManager.ShowMessage("Cannot run the command", "Only ver. 4.x primary instance that use RavenDB ver. 3.5 can be forced upgraded.");
+
+            return;
+        }
+
+        if (DotnetVersionValidator.FrameworkRequirementsAreMissing(needsRavenDB: true, out var missingMessage))
+        {
+            await windowManager.ShowMessage("Missing prerequisites", missingMessage, acceptText: "Cancel", hideCancel: true);
+            return;
+        }
+
+        var instance = InstanceFinder.FindInstanceByName<ServiceControlInstance>(model.Name);
+
+        var upgradeInfo = UpgradeControl.GetUpgradeInfoForTargetVersion(serviceControlInstaller.ZipInfo.Version, instance.Version);
+        var upgradeOptions = new ServiceControlUpgradeOptions
+        {
+            UpgradeInfo = upgradeInfo,
+        };
+
+        await UpgradeServiceControlInstance(model, instance, upgradeOptions);
+
+        await eventAggregator.PublishOnUIThreadAsync(new ResetInstances());
+    }
+
+    async Task UpgradeServiceControlInstance(ServiceControlAdvancedViewModel model, ServiceControlInstance instance, ServiceControlUpgradeOptions upgradeOptions)
+    {
+        using (var progress = model.GetProgressObject($"UPGRADING {model.Name}"))
+        {
+            var reportCard = new ReportCard();
+            var restartAgain = model.IsRunning;
+
+            var stopped = await model.StopService(progress);
+
+            if (!stopped)
+            {
+                await eventAggregator.PublishOnUIThreadAsync(new RefreshInstances());
+
+                reportCard.Errors.Add("Failed to stop the service");
+                reportCard.SetStatus();
+                await windowManager.ShowActionReport(reportCard, "ISSUES UPGRADING INSTANCE", "Could not upgrade instance because of the following errors:");
+
+                return;
+            }
+
+            reportCard = await Task.Run(() =>
+            {
+                //HINT: we wipe out the database before we continue with the upgrade
+                FileUtils.DeleteDirectory(instance.DBPath, recursive: true, contentsOnly: true);
```

Comment:
Should we try to move the directory aside instead?

Reply:
I'm good with moving the directory. I will change the message content shown to the users as well to make sure they are aware that the data is being moved.